File: kit-fastapi/tests/fakes/fake_clock.py
"""Horloge figée pour des tests déterministes."""
from __future__ import annotations

from datetime import datetime, timedelta, timezone


class FakeClock:
    def __init__(self, start: datetime | None = None) -> None:
        self._now = start or datetime(2026, 1, 1, tzinfo=timezone.utc)

    def now(self) -> datetime:
        return self._now

    def advance(self, seconds: int) -> None:
        self._now = self._now + timedelta(seconds=seconds)

File: kit-php/tests/Fakes/FakeClock.php
<?php

declare(strict_types=1);

namespace TodoApp\Tests\Fakes;

use DateTimeImmutable;
use TodoApp\Domain\Todo\Clock;

/** Horloge figée pour des tests déterministes. */
final class FakeClock implements Clock
{
    public function __construct(private DateTimeImmutable $now = new DateTimeImmutable('2026-01-01T00:00:00+00:00'))
    {
    }

    public function now(): DateTimeImmutable
    {
        return $this->now;
    }

    public function advance(string $modifier): void
    {
        $this->now = $this->now->modify($modifier);
    }
}
